virttest/virsh.py

```diff
@@ -1634,12 +1645,18 @@ def detach_device(domainarg=None, filearg=None,
     :param domain_opt: Option to --domain parameter
     :param file_opt: Option to --file parameter
     :param flagstr: string of "--force, --persistent, etc."
+    :param wait_remove_event: wait until device_remove event comes
+    :param event_timeout: timeout for virsh event command
     :param dargs: standardized virsh function API keywords
     :return: CmdResult instance
     """
-    return _adu_device("detach-device", domainarg=domainarg, filearg=filearg,
-                       domain_opt=domain_opt, file_opt=file_opt,
-                       flagstr=flagstr, **dargs)
+    detach_cmd_rv = _adu_device("detach-device", domainarg=domainarg, filearg=filearg,
+                                domain_opt=domain_opt, file_opt=file_opt,
+                                flagstr=flagstr, **dargs)
+    if wait_remove_event:
+        event(domain=domainarg, event='device-removed', event_timeout=event_timeout)
```

Comment:
sorry @qiankehan, one more thing, you missed dargs

Reply:
I want to bring one open question here: there are two ways to bring this idea into facts
way 1)update current function with more parameters(as current patch show)
way 2) create another overload function with keep original unchanged.
Provide you preference here as per your thoughts 
@kylazhang 
@dzhengfy 
@smitterl

Reply:
> sorry @qiankehan, one more thing, you missed dargs


Fixed

Reply:
> I want to bring one open question here: there are two ways to bring this idea into facts
> way 1)update current function with more parameters(as current patch show)
> way 2) create another overload function with keep original unchanged.

Another way to keep to tp-libvirt code unchanged: set **wait_remove_event=True** by default

> Provide you preference here as per your thoughts
> @kylazhang
> @dzhengfy
> @smitterl
> @Yingshun